From b0e844967728edf814d611bb1c69d320a8a6c64d Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Tue, 1 Jul 2014 15:58:16 +0200
Subject: [PATCH 3/3] functions.prf: Try to add _moc suffix

* otherwise moc generated files have the same basename and do_configure fails with:
  core_generated.gyp:QtWebEngineCore#target has several files with the same basename:
  |   javascript_dialog_controller: javascript_dialog_controller.cpp
  | /OE/build/oe-core/tmp-eglibc/work/i586-oe-linux/qtwebengine/0.9.99+gitAUTOINC+21f6ce84ec-r0/git/src/core/.moc/moc_/OE/build/oe-core/tmp-eglibc/work/i586-oe-linux/qtwebengine/0.9.99+gitAUTOI.moc+21f6ce84ec-r0/git/src/core/javascript_dialog_controller.cpp
  ...

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
---
 tools/qmake/mkspecs/features/functions.prf | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tools/qmake/mkspecs/features/functions.prf b/tools/qmake/mkspecs/features/functions.prf
index 15c2941..78b1190 100644
--- a/tools/qmake/mkspecs/features/functions.prf
+++ b/tools/qmake/mkspecs/features/functions.prf
@@ -40,8 +40,8 @@ defineReplace(findIncludedMocFiles) {
 defineReplace(mocOutput) {
   out = $$1
   # The order is important, since the output of the second replace would end up accidentaly transformed by the first one
-  out = $$replace(out, ^(.*)($$join(QMAKE_EXT_CPP,|))$, $${QMAKE_CPP_MOD_MOC}\\1$${QMAKE_EXT_CPP_MOC})
-  out = $$replace(out, ^(.*)($$join(QMAKE_EXT_H,|))$, $${QMAKE_H_MOD_MOC}\\1$${first(QMAKE_EXT_CPP)})
+  out = $$replace(out, ^(.*)($$join(QMAKE_EXT_CPP,|))$, $${QMAKE_CPP_MOD_MOC}\\1_moc$${QMAKE_EXT_CPP_MOC})
+  out = $$replace(out, ^(.*)($$join(QMAKE_EXT_H,|))$, $${QMAKE_H_MOD_MOC}\\1_moc$${first(QMAKE_EXT_CPP)})
   return($$out)
 }
 
-- 
2.1.1

